Rules/UseUsingScopeModifierInNewRunspaces.cs

```diff
@@ -0,0 +1,413 @@
+    /// </summary>
+#if !CORECLR
+[Export(typeof(IScriptRule))]
+#endif
+    public class UseUsingScopeModifierInNewRunspaces : IScriptRule
+    {
+        /// <summary>
+        /// AnalyzeScript: Analyzes the ast to check variables in script blocks that will run in new runspaces are properly initialized or passed in with $using:
+        /// </summary>
+        /// <param name="ast">The script's ast</param>
+        /// <param name="fileName">The script's file name</param>
+        /// <returns>A List of results from this rule</returns>
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            
+            var visitor = new SyntaxCompatibilityVisitor(this, fileName);
+            ast.Visit(visitor);
+            return visitor.GetDiagnosticRecords();
+        }
+
+        /// <summary>
+        /// GetName: Retrieves the name of this rule.
+        /// </summary>
+        /// <returns>The name of this rule</returns>
+        public string GetName()
+        {
+            return string.Format(CultureInfo.CurrentCulture, Strings.NameSpaceFormat, GetSourceName(),
+                Strings.UseUsingScopeModifierInNewRunspacesName);
+        }
+
+        /// <summary>
+        /// GetCommonName: Retrieves the common name of this rule.
+        /// </summary>
+        /// <returns>The common name of this rule</returns>
+        public string GetCommonName()
+        {
+            return string.Format(CultureInfo.CurrentCulture, Strings.UseUsingScopeModifierInNewRunspacesCommonName);
+        }
+
+        /// <summary>
+        /// GetDescription: Retrieves the description of this rule.
+        /// </summary>
+        /// <returns>The description of this rule</returns>
+        public string GetDescription()
+        {
+            return string.Format(CultureInfo.CurrentCulture, Strings.UseUsingScopeModifierInNewRunspacesDescription);
+        }
+
+        /// <summary>
+        /// GetSourceType: Retrieves the type of the rule: builtin, managed or module.
+        /// </summary>
+        public SourceType GetSourceType()
+        {
+            return SourceType.Builtin;
+        }
+
+        /// <summary>
+        /// GetSeverity: Retrieves the severity of the rule: error, warning of information.
+        /// </summary>
+        /// <returns></returns>
+        public RuleSeverity GetSeverity()
+        {
+            return RuleSeverity.Warning;
+        }
+
+        /// <summary>
+        /// GetSourceName: Retrieves the module/assembly name the rule is from.
+        /// </summary>
+        public string GetSourceName()
+        {
+            return string.Format(CultureInfo.CurrentCulture, Strings.SourceName);
+        }
+    }
+
+#if !(PSV3 || PSV4)
+    internal class SyntaxCompatibilityVisitor : AstVisitor2
+#else
+        private class SyntaxCompatibilityVisitor : AstVisitor

```

Comment:
You want this to be an `internal` rather than a `private` class. Alternatively you can move it inside `UseUsingScopeModifierInNewRunspaces` and make it private. But this is why PS3 compile is failing

Reply:
That was fast, thank you! Came here to say I found exactly this :-)

Reply:
Build now fails on 
```
Executing script C:\Users\Jos.Koelewijn\github\PSScriptAnalyzer\Tests\Rules\UseCompatibleCommands.Tests.ps1

  Describing UseCompatibleCommands

    Context Targeting a single profile

    Context Checking a file against many targets

    Context Checking PSSA repository scripts
      [-] Ensures that PSSA build scripts are cross compatible with everything 352ms
        Expected 0, because PSUseCompatibleCommands, PSUseCompatibleCommands, but got 2.
        336:             $diagnostics.Count | Should -Be 0 -Because ($diagnostics.RuleName -join ', ')
        at <ScriptBlock>, C:\Users\Jos.Koelewijn\github\PSScriptAnalyzer\Tests\Rules\UseCompatibleCommands.Tests.ps1: line 336
```

Apparently something I did affected how `PSUseCompatibleCommands` and `PSUseCompatibleCommands` are evaluated here.

Reply:
I'll have to see about this later.